n2y/converter.py

```diff
@@ -384,46 +385,73 @@ def to_pandoc(self):
 
 
 class ImageBlock(Block):
+    # `ImageBlock` Class Object Initialization
     def __init__(self, client: Client, block, get_children=True):
+        # Initialize Inherited `Block` Class
         super().__init__(client, block, get_children)
         self.file = File(block['image'])
 
+    # Returns Block Information In Pandoc Format
     def to_pandoc(self):
-        url = None
-        if self.file.type == "external":
-            url = self.file.url
-        elif self.file.type == "file":
-            url = self.file.download()
         caption = RichTextArray(self.caption)
+        external = self.file.type == "external"
+        file = self.file.type == "file"
+
+        if external:
+            url = self.file.create_unique_url(self.file.url)
+        elif file:
+            url = self.file.download()
+
         return Para([Image(('', [], []), caption.to_pandoc(), (url, ''))])
 
 
 class File():
+    # `File` Class Object Initialization
     def __init__(self, obj):
-        if obj['type'] == "file":
-            self.type = "file"
-            self.url = obj['file']['url']
-            self.expiry_time = obj['file']['expiry_time']
-        elif obj['type'] == "external":
-            self.type = "external"
-            self.url = obj['external']['url']
-
+        # Set Values For The `type` And `url` Keys Based On That Of `obj` Param
+        for type in ["file", "external"]:
+            if obj['type'] == type:
+                self.type = type
+                self.url = obj[type]['url']
+                # Only "file" Typed `obj`s Have `expiry_time`, So Set That Key If It's There
+                self.expiry_time = obj[type]['expiry_time'] if type == "file" else None
+
+    # Pevents Collisions In The File Paths N2Y Creates
+    def create_unique_url(self, url):
```

Comment:
Does this function create unique URLs or unique file names? If the former is the case, I don't see how/why we would want to create unique URLs.

Reply:
I changed the name